### PR TITLE
Validate and createValidationRule methods

### PR DESCRIPTION
**Pull request by alackmann**
Added support for the existing API methods:

* /vouchers/{code}/validate - determine if a code is valid and retrieve metadata
* /validation-rules/ - add validation rules to an existing code

### CODE REVIEW

File: VoucherifyClient.php [hunk 1]
```diff
@@ -143,6 +143,25 @@ public function redeem($code, $trackingId) {
...
 
         /**
+         * @param string|array $code Voucher code or array with voucher (code), customer profile and order amount
+         * @param string|null $trackingId Provided tracking id
+         *
+         * Validate voucher
+         *
+         * @throws Voucherify\ClientException
+         */
+        public function validate($code, $trackingId = NULL) {
```
Comment (mandraszyk) [inline VoucherifyClient.php]: validate function signature should be `validate ($code, $context)` where code is a string and context is an array or object.

File: VoucherifyClient.php [hunk 2]
```diff
@@ -143,6 +143,25 @@ public function redeem($code, $trackingId) {
...
+            $context = array();
+            if (is_array($code)) {
+                $context = $code;
+                $code = $context["voucher"];
+                unset($context["voucher"]);
+            }
+
+            return $this->apiRequest("POST", "/vouchers/" . urlencode($code) . "/validate", ["tracking_id" => $trackingId], $context);
```
Comment (mandraszyk) [inline VoucherifyClient.php]: `$tracking_id` should be sent as `$context.customer.source_id` - please check updated documentation:
https://docs.voucherify.io/reference#validate-voucher

Comment (mandraszyk) [inline VoucherifyClient.php]: `$context` should be mapped to object - sending empty array will end with error: { code:400 message: "The request entity was empty" } - `$context.customer` and `$context.order` are optional

File: VoucherifyRequest.php [hunk 3]
```diff
@@ -88,6 +88,10 @@ protected function apiRequest($method, $endpoint, $params, $data) {
             throw new ClientException($error);
         } // Invalid status code
         else if ($statusCode >= 400) {
+            $resultDecoded = json_decode($result, true);
```
Comment (mandraszyk) [inline VoucherifyRequest.php]: please leave current error response format - we will update ClientException in next iteration to match API error format { code, message, detail }

**Comment by tpindel** [comment]
@alackmann Thank you very much for this contribution. Please apply enhancements to @mandraszyk 's comments and afterwards we will merge this PR.

FYI We are going to release new version of this SDK where better error handling is included and other API methods covered. I'm sure you will find it useful.

**Comment by mandraszyk** [comment]
@alackmann Hi, we have made some changes in client structure. Your PR cannot be merged now. In near future we will add methods from missing namespaces: validations, distributions, products and segments.

**Pull request closed without merging**